core/src/sigverify_stage.rs

```diff
@@ -266,6 +273,21 @@ impl SigVerifyStage {
         }
     }
 
+    fn maybe_shrink_batches(packet_batches: &mut Vec<PacketBatch>) -> (u64, usize) {
+        let mut shrink_time = Measure::start("sigverify_shrink_time");
+        let num_packets = count_packets_in_batches(packet_batches);
+        let num_discarded_packets = count_discarded_packets(packet_batches);
```

Comment:
would be better to do this in one pass. Also we probably already know the answer from previous steps.